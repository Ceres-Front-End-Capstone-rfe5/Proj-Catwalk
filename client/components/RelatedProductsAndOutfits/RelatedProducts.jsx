import React from 'react';
import axios from 'axios';
import Product from './ProductCard';
class RelatedProducts extends React.Component {
constructor(props) {
  super(props);
  this.state = {
    productsShown: [],
    hiddenLeft: [],
    hiddenRight: [],
    left: false,
    right: true
  }
}

render() {
  const {currentProduct, currentProductId, products, relatedProducts} = this.props;

  //console.log(this.props)
  return (
    <div className = "related-products-car">
      <h2> Related sub</h2>
      <Product relatedProducts={relatedProducts} currentProductId={currentProductId} products={products} currentProduct={currentProduct}></Product>
    </div>
  )
}
}

export default RelatedProducts;