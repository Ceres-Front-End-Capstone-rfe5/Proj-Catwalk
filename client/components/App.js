
import React from "react";
import ProductDetail from './ProductDetail';
import RelatedProductsAndOutfits from './RelatedPoductsAndOutfit.jsx';
import ReviewsCore from './ReviewsCore.jsx';
import axios from 'axios';

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      products: [],
      currentProduct: {},
      currentProductId: null,
      favoritedMain: false,
    }
    this.handleRelatedProductsClick = this.handleRelatedProductsClick.bind(this)
    this.handleAddMainAsFavorite = this.handleAddMainAsFavorite.bind(this)
  }

  componentDidMount() {
    const productId = '37313';
    axios.get('/api/products').then((products) => {
      axios.get(`/api/products/${productId}`).then((product) => {
        this.setState({
          currentProduct: product.data,
          currentProductId: product.data.id,
          products: products.data,
        })
      })
    })
    if(sessionStorage.yourOutfits && sessionStorage.yourOutfits.split(',').includes(this.state.currentProductId+'')) {
      this.setState({
        favoritedMain: true
      })
    } else {
      this.setState({
        favoritedMain: false
      })
    }
  }
//   componentDidUpdate(prevState) {
//     if(prevProps.currentProduct !== this.state.currentProduct && sessionStorage.yourOutfits && sessionStorage.yourOutfits.split(',').includes(this.state.currentProductId+'')) {
//       console.log(prevProps)
//       this.setState({favoritedMain: true})
//   }
// }

  handleAddMainAsFavorite(bool) {
    this.setState({favoritedMain: bool})
  }

  handleRelatedProductsClick(id) {
    axios.get(`/api/products/${id}`)
    .then(product => {
      this.setState({
        currentProduct: product.data,
        currentProductId: product.data.id,
        favoritedMain: false
      });
    })
  }

  render() {
    const { currentProduct, currentProductId, products, favoritedMain} = this.state;
    console.log('APP STATE FAVE MAIN', favoritedMain)
    return (
      <>
        <ProductDetail favoritedMain={favoritedMain} currentProduct={currentProduct} handleAddMainAsFavorite={this.handleAddMainAsFavorite} />
        <br></br>
        <RelatedProductsAndOutfits currentProduct={currentProduct} products={products} currentProductId={currentProductId} handleRelatedProductsClick={this.handleRelatedProductsClick} favoritedMain={favoritedMain}/>
        <br></br>
        {/* <ReviewsCore currentProductId={currentProductId} reviewList={this.state.reviews} /> */}
      </>
    );
  }
}

export default App;

