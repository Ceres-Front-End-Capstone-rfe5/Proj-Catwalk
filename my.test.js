/**
 * @jest-environment jsdom
 */


// test('adds 1 + 2 to equal 3', () => {
//     expect(1+2).toBe(3);
//   });



import React from 'react';

import App from './client/components/App.js'
import Adapter from 'enzyme-adapter-react-16';
import Enzyme, {shallow} from 'enzyme'
import RelatedProducts from './client/components/RelatedProductsAndOutfits/RelatedProducts.jsx'
Enzyme.configure({ adapter: new Adapter() })


test('Renders Correctly', () => {
  const app = shallow(<App currentProductId='37313'/>)
  //console.log(app.currentProductId)
  expect(app).toMatchSnapshot();
})


// const clickFn = jest.fn();

// test('Re-renders App when clicking on related product correctly', () => {
//   const card = shallow(<RelatedProducts  onClick={clickFn}/>)
//   var nextButton = card.find('.carousal-next')
//   console.log(nextButton)
//   nextButton.simulate('click');

//   expect(clickFn).toHaveBeenCalled();
// })