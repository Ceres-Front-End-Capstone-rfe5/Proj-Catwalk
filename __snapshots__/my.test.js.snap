// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Correctly 1`] = `ShallowWrapper {}`;

exports[`Updates current product when rendering page 1`] = `ShallowWrapper {}`;
